### PR TITLE
[BUG][JAVA][SPRING] Fix serialization when there is a discriminator with mapping

### PR DESCRIPTION
**Pull request by jorgerod**
Closes: #14731 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09)

### CODE REVIEW

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.fasterxml.jackson.databind.Module;
+import org.openapitools.jackson.nullable.JsonNullableModule;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+
+@SpringBootApplication
+@ComponentScan(basePackages = {"org.openapitools", "org.openapitools.api" , "org.openapitools.configuration"})
+public class OpenApiGeneratorApplication {
```
Comment (welshm) [inline samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java:12]: Any idea why this is being created? I wouldn't imagine this should exist from your change

**Review by welshm — COMMENTED** [review]
Change itself looks fine, but some of the samples being created don't seem related to this change

File: samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java (hunk 1 above)
Comment (jorgerod) [reply samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java:12]: I have no idea.  I have simply executed the test script and they have been generated.  

what should I do? Remove It?

Comment (welshm) [reply samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java:12]: If you regenerate on `master` does it also output these files?

Comment (jorgerod) [reply samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/OpenAPI2SpringBoot.java]: I deleted those examples and relaunched the script and it worked correctly.
It is not clear to me what happened before.

**Review by borsch — APPROVED** [review]

**Review by welshm — APPROVED** [review]

**Pull request merged**